### DISCUSSION TITLE
Diagnostics Disabled flag seems do not work

### DISCUSSION THREAD

**Discussion by LLMChild** [nvim-tree/nvim-tree.lua#2792] [Q&A]
https://github.com/nvim-tree/nvim-tree.lua/blob/26632f496e7e3c0450d8ecff88f49068cecc8bda/lua/nvim-tree/diagnostics.lua#L44

I am trying to write a keybind to turn off diagnostics.
```lua
vim.g.diagnostics_active = true
function _G.toggle_diagnostics()
if vim.g.diagnostics_active then
  vim.g.diagnostics_active = false
  vim.diagnostic.hide()
  vim.lsp.handlers["textDocument/publishDiagnostics"] = function() end
  vim.notify("Diagnostics are now off", vim.log.levels.INFO, { title = "Diagnostics" })
else
  vim.g.diagnostics_active = true
  vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(
    vim.lsp.diagnostic.on_publish_diagnostics, {
      virtual_text = false,
      signs = true,
      underline = true,
      update_in_insert = true,
    }
  )
  vim.notify("Diagnostics are now no", vim.log.levels.INFO, { title = "Diagnostics" })
end
end
map('n', '<leader>td', ':call v:lua.toggle_diagnostics()<CR>',  {noremap = true, silent = true, desc = "Toggle Diagnostics"})
```
This works for my diagnostics all buffers but does not work for nvim-tree.
So I am trying to find a function which can toggle Diagnostics for nvim-tree in runtime.
Sadly, nvim-tree does not have anything like this.
So I checked the code , and I found this flag.
Then I add enable and enable(false) in my keybindings.
like this 
```lua
function _G.toggle_diagnostics()
if vim.g.diagnostics_active then
  vim.g.diagnostics_active = false
  vim.diagnostic.hide()
  vim.diagnostic.enable(false)
  vim.lsp.handlers["textDocument/publishDiagnostics"] = function() end
  vim.notify("Diagnostics are now off", vim.log.levels.INFO, { title = "Diagnostics" })
else
  vim.g.diagnostics_active = true
  vim.diagnostic.enable()
  vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(
    vim.lsp.diagnostic.on_publish_diagnostics, {
      virtual_text = false,
      signs = true,
      underline = true,
      update_in_insert = true,
    }
  )
```
It didn't not work either, and I checked status using `:lua print(vim.diagnostic.is_disabled())`.
But nvim-tree still won't hide the diagnostic.
Can anybody in this community help me?

**Comment by LLMChild** [comment]
I also checked  `vim.fn.has "nvim-0.9"==1` is true

**Reply by alex-courtis** [reply]
We should not receive any new diagnostics info when diagnostics has been disabled - no `DiagnosticChanged` events will be published.

What do you see after you disable nvim diagnostics? I imagine that the old values would persist forever.

There doesn't seem to be an event; do you know of another plugin that handles diagnostic disable?

**Reply by LLMChild** [reply]
After I disable nvim diagnostics, there is not new diagnostics events anymore.
But all these old diagnostics signs still keeps on the nvim-tree. 
I think this means `vim.diagnostics.hide` or `vim.diagnostics.disable` would not clear the previous diagnostics signs of nvim-tree.
But they can hide diagnostic of current buffer.
With `vim.diagnostics.reset`, I get what i want. It can clear all old signs of nvim-tree.
But if I want to toggle diagnostics again, I need to run it again.
I want something like `vim.diagnostics.hide` to hide all these old signs of nvim-tree and I would like to know how to do such thing.

**Reply by alex-courtis** [reply]
`:help diagnostic.txt` doesn't explicitly cover the conditions that will fire the event. It does look like `reset` achieves the functionality you are after as it fires the event that `disable` does not.

I might be misunderstanding: could you simply call `reset` after `hide` or `disable`? My testing showed that calling `reset` and `disable` in any order achieves the desired functionality.

You can use `hide`, `enable` and `reset` for specific buffers and namespaces if you wish.
